src/pages/main/components/MapModal.tsx

```diff
@@ -1,106 +1,266 @@
-                      <Placemark
-                        modules={['geoObject.addon.balloon']}
-                        key={accommodation.id}
-                        geometry={[accommodation.address.latitude, accommodation.address.longitude]}
-                        options={{ preset: 'islands#violetIcon' }}
-                        properties={{
-                          iconCaption: accommodation.price + '$',
-                          balloonContentBody: `
-                            <a href="${originUrl}/rooms/${accommodation.id}" style="overflow: hidden; position: relative; text-decoration: none; color: #000; width: 250px;     height: 150px;"  >
-                              <div style="padding: 16px 0; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); font-size: 16px; max-width: 250px;">
-                                <div style="font-weight: bold; font-size: 20px; margin-bottom: 8px;">
-                                  ${accommodation.title}
-                                </div>
-                                <img src="${accommodation.thumbnailUrl}" alt="" onerror="this.onerror=null; this.src='/src/assets/no-image.jpg';"  style="width: 250px; height: 150px; box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1); margin-bottom: 8px;border-radius: 10px;  max-height: 150px; object-fit: cover;" />
-                                <Typography mt={2}>
-                                  ${accommodation.address.city}, ${accommodation.address.country}
-                                </Typography>
-                              </div>
-                            </a>
-                          `,
-                        }}
-                      />
-                    ))}
-                  </Clusterer>
-                </Map>
-              </div>
-            </YMaps>
-          </Box>
-        </Box>
-      </Fade>
-    </Modal>
-  );
-}
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+
+import CloseIcon from '@mui/icons-material/Close';
+import { Box, Button, Fade, Modal, Stack, Typography } from '@mui/material';
+import { useCallback, useEffect } from 'react';
+
+import { MapModalProps } from '@src/types/accommodation.types';
+import { modalStyles } from './Modal.styles';
+
+const maxAccommInMap = import.meta.env.VITE_MAX_ACCOMMODATIONS_IN_MAP as string;
+const originUrl = import.meta.env.VITE_ORIGIN_URL as string;
+
+export default function MapModal({ open, setOpen, searchParamsAsObject }: MapModalProps) {
+  const { location } = searchParamsAsObject;
+
+  const handleClose = useCallback(() => setOpen(false), [setOpen]);
+
+  function initMapWithCoordinates(coords) {
+    ymaps.ready(() => {
+      const mapContainer = document.getElementById('map');
+      if (mapContainer && mapContainer.children.length > 0) {
+        return;
+      }
+      init(coords);
+    });
+  }
+
+  useEffect(() => {
+    if (location.trim() === '') {
+      initMapWithCoordinates([55.76, 37.64]);

```

Comment:
as i remember we have already constant with initial coordinates, If I'm not mistaken @zilola08 knows